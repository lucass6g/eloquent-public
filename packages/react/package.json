{
  "name": "@eloquent/react",
  "version": "1.0.0",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup",
    "clean": "rm -rf dist && rm -rf node_modules"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.1.0",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "react": "^18.3.1",
    "typescript": "^5.4.5"
  }
}
