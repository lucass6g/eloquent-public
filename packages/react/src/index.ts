export {Switch, Case, Default} from './components/conditional-switch'