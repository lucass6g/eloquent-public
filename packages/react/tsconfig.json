{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
