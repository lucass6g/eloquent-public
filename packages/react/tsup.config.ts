import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})