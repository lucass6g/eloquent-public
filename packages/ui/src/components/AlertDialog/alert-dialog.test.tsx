import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { AlertDialog } from '.';

const mockComponent = <AlertDialog.Root>
    <AlertDialog.Trigger data-testid="trigger">
        Abrir
    </AlertDialog.Trigger>
    <AlertDialog.Content>
        <AlertDialog.Header>
            <AlertDialog.Title data-testid="title">
                Você está certo disso?
            </AlertDialog.Title>
            <AlertDialog.Description data-testid="description">
                Esta ação não pode ser desfeita. Isso excluirá permanentemente sua conta e removerá seus dados dos nossos servidores
            </AlertDialog.Description>
        </AlertDialog.Header>
        <AlertDialog.Footer>
            <AlertDialog.Cancel data-testid="cancel">
                Cancelar
            </AlertDialog.Cancel>
            <AlertDialog.Action data-testid="Action">
                Continuar
            </AlertDialog.Action>
        </AlertDialog.Footer>
    </AlertDialog.Content>
</AlertDialog.Root>

describe('AlertDialog Component', () => {
    it('should open and display dialog content when trigger is clicked', () => {
        const component = render(mockComponent);
        const triggerButton = screen.getByTestId('trigger');
        fireEvent.click(triggerButton);
        const title = screen.getByTestId('title');
        const description = screen.getByTestId('description');
        const cancelButton = screen.getByTestId('cancel');
        const actionButton = screen.getByTestId('Action');
        expect(title).toBeInTheDocument();
        expect(title).toHaveTextContent('Você está certo disso?');
        expect(description).toBeInTheDocument();
        expect(description).toHaveTextContent('Esta ação não pode ser desfeita. Isso excluirá permanentemente sua conta e removerá seus dados dos nossos servidores');
        expect(cancelButton).toBeInTheDocument();
        expect(actionButton).toBeInTheDocument();
        expect(component.baseElement).toMatchSnapshot();
    });
    it('should close the dialog when the cancel button is clicked', () => {
        const component = render(mockComponent);
        const triggerButton = screen.getByTestId('trigger');
        fireEvent.click(triggerButton);
        const cancelButton = screen.getByTestId('cancel');
        fireEvent.click(cancelButton);
        const title = screen.queryByTestId('title');
        expect(title).not.toBeInTheDocument();
        expect(component.baseElement).toMatchSnapshot();
    });
});