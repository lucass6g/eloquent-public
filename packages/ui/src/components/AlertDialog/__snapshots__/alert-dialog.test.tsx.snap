// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AlertDialog Component > should close the dialog when the cancel button is clicked 1`] = `
<body
  style=""
>
  <div>
    <button
      aria-controls="radix-:r3:"
      aria-expanded="false"
      aria-haspopup="dialog"
      data-state="closed"
      data-testid="trigger"
      type="button"
    >
      Abrir
    </button>
  </div>
</body>
`;

exports[`AlertDialog Component > should open and display dialog content when trigger is clicked 1`] = `
<body
  data-scroll-locked="1"
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <button
      aria-controls="radix-:r0:"
      aria-expanded="true"
      aria-haspopup="dialog"
      data-state="open"
      data-testid="trigger"
      type="button"
    >
      Abrir
    </button>
  </div>
  <div
    aria-hidden="true"
    class="fixed inset-0 z-50 bg-black/80 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto;"
  />
  <div
    aria-describedby="radix-:r2:"
    aria-labelledby="radix-:r1:"
    class="fixed left-[50%] top-[50%] z-50 grid lg:w-full max-w-lg min-w-[328px] translate-x-[-50%] translate-y-[-50%] gap-4 border bg-background p-6 shadow-lg duration-200 data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[state=closed]:slide-out-to-left-1/2 data-[state=closed]:slide-out-to-top-[48%] data-[state=open]:slide-in-from-left-1/2 data-[state=open]:slide-in-from-top-[48%] rounded-[8px]"
    data-state="open"
    id="radix-:r0:"
    role="alertdialog"
    style="pointer-events: auto;"
    tabindex="-1"
  >
    <div
      class="flex flex-col space-y-2 text-center sm:text-left"
    >
      <h2
        class="text-lg font-semibold text-green-dark-600"
        data-testid="title"
        id="radix-:r1:"
      >
        Você está certo disso?
      </h2>
      <p
        class="text-sm text-neutral-500"
        data-testid="description"
        id="radix-:r2:"
      >
        Esta ação não pode ser desfeita. Isso excluirá permanentemente sua conta e removerá seus dados dos nossos servidores
      </p>
    </div>
    <div
      class="flex flex-col-reverse sm:flex-row sm:justify-end sm:space-x-2"
    >
      <button
        class="border border-border text-secondary-foreground text-sm font-medium py-2 px-4 rounded-[360px] hover:bg-secondary-hover disabled:text-neutral-500 mt-2 sm:mt-0"
        data-testid="cancel"
        type="button"
      >
        Cancelar
      </button>
      <button
        class="bg-primary text-sm font-medium text-primary-foreground shadow py-2 px-4 rounded-[360px] hover:bg-primary-hover disabled:bg-neutral-300 disabled:text-neutral-500 disabled:font-medium"
        data-testid="Action"
        type="button"
      >
        Continuar
      </button>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;
